addons/website/models/ir_ui_view.py

```diff
@@ -24,12 +24,17 @@ class View(models.Model):
     website_id = fields.Many2one('website', ondelete='cascade', string="Website")
     page_ids = fields.One2many('website.page', 'view_id')
     first_page_id = fields.Many2one('website.page', string='Website Page', help='First page linked to this view', compute='_compute_first_page_id')
+    has_duplicate_key = fields.Boolean(compute='_compute_has_duplicate_key')
 
     @api.multi
     def _compute_first_page_id(self):
         for view in self:
             view.first_page_id = self.env['website.page'].search([('view_id', '=', view.id)], limit=1)
 
+    def _compute_has_duplicate_key(self):
+        for view in self:
+            view.has_duplicate_key = self.search_count([('key', '=', view.key), ('website_id', '=', view.website_id.id)]) > 1
```

Comment:
Maybe it's possible to use a `group_by` and avoid a `search` in a loop?

Reply:
Premature optimization imho, but I can do it